packages/grant-explorer/src/features/projects/hooks/useProject.ts

```diff
@@ -0,0 +1,18 @@
+import useSWR from "swr";
+import { DataLayer } from "data-layer";
+import { AlloVersion } from "data-layer/src/data-layer.types";
+
+type Params = {
+  projectId?: string;
+};
+
+export function useProject(params: Params, dataLayer: DataLayer) {
+  const shouldFetch = Object.values(params).every(Boolean);
+  return useSWR(shouldFetch ? ["applications", params] : null, async () => {
+    const validatedParams = {
+      projectId: params.projectId as string,
+      alloVersion: "allo-v2" as AlloVersion,
```

Comment:
let's use `as const` instead of `as AlloVersion`.

with `as const` it ensures it's a valid value. using `allo-v3` will fail at compile time. 
with `as AlloVersion` any string will work.

Reply:
@gravityblast `type AlloVersion = "allo-v1" | "allo-v2";`
Why would with `as AlloVersion` any string will work?

Reply:
@bhargavaparoksham  cause it converted to that type. try putting `"allo-v3" as AlloVersion` and it still works.
Instead `"allo-v3" as const` would fail at compile time

Reply:
Aah got it.